Simplenote/src/main/java/com/automattic/simplenote/widgets/SimplenoteEditText.java

```diff
@@ -191,28 +191,45 @@ public void run() {
         }
     }
 
-    // Returns the line position of the text cursor
-    private int getCurrentCursorLine() {
-        int selectionStart = getSelectionStart();
-        Layout layout = getLayout();
-
-        if (!(selectionStart == -1)) {
-            return layout.getLineForOffset(selectionStart);
+    /**
+     *  Finds the start of line containing the start of selection
+     */
+    private int findStartOfLineOfSelection() {
+        int position = getSelectionStart();
+        if(position == -1) {
+            position = 0;
+        }
+        Editable editable = getText();
+        for (int i = position - 1; i >= 0; i--) {
+            if (editable.charAt(i) == '\n') {
+                return i + 1;
+            }
         }
-
         return 0;
     }
 
-    public void insertChecklist() {
-        int start, end;
-        int lineNumber = getCurrentCursorLine();
-        start = getLayout().getLineStart(lineNumber);
-
-        if (getSelectionEnd() > getSelectionStart() && !selectionIsOnSameLine()) {
-            end = getSelectionEnd();
-        } else {
-            end = getLayout().getLineEnd(lineNumber);
+    /**
+     *  Finds the end of line containing the end of selection
+     */
+    private int findEndOfLineOfSelection() {
+        int position = getSelectionEnd();
+        if(position == -1) {
+            position = 0;
+        }
```

Comment:
```suggestion
        // Since the minimum position index is 0 and the selection end can return -1 for no
        // selection, get the maximum value between 0 and getSelectionEnd().
        int position = Math.max(0, getSelectionEnd());
```

We can use `Math.max` as a trick similar to the `return` statement below.  It may not be obvious why we're checking for `-1` if someone isn't familiar with `getSelectionEnd()`.  So, we may want to add a comment explaining what's happening.

It's debatable whether using `Math.max` or `if (position == -1)` is better.  Changing this to `Math.max` is not necessary.  It's just a suggestion.  I wanted to point out a possible alternative in case you want to use it.  Either way, we should probably add some comments so that the logic is clear.

Reply:
Done 8ece2a016764255da072bfa2a10d088c478be3cd